src/Scripting/ImGuiBindings.cpp

```diff
@@ -0,0 +1,527 @@
+#include "ImGuiBindings.h"
+
+#include <imgui/imgui.h>
+#include <string>
+#include <vector>
+#include <tuple>
+#include <sol/sol.hpp>
+
+// Windows
+bool imGuiBegin(const std::string &name) { return ImGui::Begin(name.c_str()); }
+std::tuple<bool, bool> imGuiBegin(const std::string &name, bool isOpened, int flags) {
```

Comment:
You can just name these differently so that you won't need `sol::resolve`.

`imGuiBegin` / `imGuiBeginEx`, for example.

Reply:
I do prefer to keep `sol::resolve` because it is mimicking what ImGui does with the ImGui::Begin function. It has different overloaded version with the same name ( actually more than 2 but we use just those version right now ).

Reply:
I'm saying that you will be able to use `sol::overload` w/o `sol::resolve`. So the lua side of the API will look the same as `ImGui::Begin`, only the function wrappers in this file will have different names.

Reply:
Ooh that way, yes. Fair. I'll do that with the other methods too so it will be more concise there as well.